apps/common/controllers/postcode-loop.js

```diff
@@ -70,33 +54,35 @@ module.exports = class PostcodeLoopController extends BaseController {
   }
 
   removeItem(req, res) {
-    const items = req.sessionModel.get(`${this.field}Addresses`);
-    req.sessionModel.set(`${this.field}Addresses`, _.omit(items, req.params.id));
-    const step = _.size(items) > 1 ? `/${this.field}-add-another-address` : `/${this.field}-postcode`;
+    const items = req.sessionModel.get(`${this.options.locals.field}Addresses`);
+    req.sessionModel.set(`${this.options.locals.field}Addresses`,
+      _.omit(items, req.params.id));
+    const step = _.size(items) > 1 ? `/${this.options.locals.field}-add-another-address` :
+      `/${this.options.locals.field}-postcode`;
```

Comment:
Again, this code is not responsible for removing an item so should not be in this method.
Also, it happens more than once, could be pushed upstream to parent class.